src/Aspire.Dashboard/Model/ResourceViewModel.cs

```diff
@@ -46,28 +46,56 @@ public static string GetResourceName(ResourceViewModel resource, IEnumerable<Res
     }
 }
 
-public sealed class EnvironmentVariableViewModel(string name, string? value, bool fromSpec)
+public sealed class EnvironmentVariableViewModel
 {
-    public string Name { get; } = name;
-    public string? Value { get; } = value;
-    public bool FromSpec { get; } = fromSpec;
+    public string Name { get; }
+    public string? Value { get; }
+    public bool FromSpec { get; }
 
     public bool IsValueMasked { get; set; } = true;
+
+    public EnvironmentVariableViewModel(string name, string? value, bool fromSpec)
+    {
+        ArgumentException.ThrowIfNullOrWhiteSpace(name);
+
+        Name = name;
+        Value = value;
+        FromSpec = fromSpec;
+    }
 }
 
-public sealed class EndpointViewModel(string endpointUrl, string proxyUrl)
+public sealed class EndpointViewModel
 {
-    public string EndpointUrl { get; } = endpointUrl;
-    public string ProxyUrl { get; } = proxyUrl;
+    public string EndpointUrl { get; }
+    public string ProxyUrl { get; }
```

Comment:
```suggestion
    public string EndpointUrl { get; init; }
    public string ProxyUrl { get; init; }
```

Reply:
If we make these `init` properties then the validation could be bypassed.

Reply:
Oh, how would it get bypassed?